src/linear.jl

```diff
@@ -69,18 +69,12 @@ function LinearStateSpaceProblem(
     )
 end
 
-function linear_likelihood(
-    prob::LinearStateSpaceProblem, 
-    ::LinearGaussian,
-    z_t,
-    ::Nothing
-)
-    return 0df0ads0fdsa
-end
 
-function SciMLBase.solve(
+CommonSolve.init(::LinearStateSpaceProblem, args...; kwargs...) = LinearGaussian()
+solve!(::LinearGaussian) = StateSpaceSolution(missing,missing,missing,missing) # TODO: Need more sensible defaults
+
+function CommonSolve.solve!(
     prob::LinearStateSpaceProblem{isinplace, Atype, Btype, Ctype, wtype, Rtype, utype, ttype, otype}, 
-    ::LinearGaussian, 
     args...; 
     kwargs...
 ) where {isinplace, Atype, Btype, Ctype, wtype, Rtype<:AbstractMatrix, utype, ttype, otype<:Nothing}
```

Comment:
Pretty sure this is not the right implementation of the CommonSolve interface. Should it instead be `init` passes the problem to the solver and then `solve!` is called on the solver, i.e. I'd use something like `alg.prob.stuff` to get fields from the problem type?

Reply:
`init` passes to a `cache` or `integrator` object on which `solve!` is called.

Reply:
This allows for reusing the same cache across different solves if someone `reinit!` to a new `u0` or parameters.